airflow/task/task_runner/standard_task_runner.py

```diff
@@ -84,12 +85,16 @@ def _start_by_fork(self):
             if job_id is not None:
                 proc_title += " {0.job_id}"
             setproctitle(proc_title.format(args))
-
             return_code = 0
             try:
                 # parse dag file since `airflow tasks run --local` does not parse dag file
                 dag = get_dag(args.subdir, args.dag_id)
-                args.func(args, dag=dag)
+                with _airflow_parsing_context_manager(
```

Comment:
Shouldn't the `dag = get_dag()` call be inside the context too?

Reply:
That quite depends on the answer of whether we can somewhat standardize the level at which we inject the parser (see above).

Reply:
Right. Now yes. I see we  now parse it here.

Reply:
Stupid me.